File: Projects/Dagger/Projects/Project-4-Qualifiers/app/src/main/java/com/droid/code/data/remote/NetworkService.java
package com.droid.code.data.remote;

import android.content.Context;

import com.droid.code.di.qualifiers.NetworkInfo;

import javax.inject.Inject;

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
public class NetworkService {

    private Context context;
    private String apiKey;

    @Inject
    public NetworkService(Context context, @NetworkInfo String apiKey) {
        // do the initialisation here
        this.context = context;
        this.apiKey = apiKey;
    }

    public String getDummyData() {
        return "NETWORK_DUMMY_DATA";
    }
}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/data/remote/NetworkService.kt
package com.demo.kotlin.sample.data.remote

import android.content.Context

import com.demo.kotlin.sample.di.ApplicationContext
import com.demo.kotlin.sample.di.NetworkInfo

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
@Singleton
class NetworkService @Inject constructor(
        @ApplicationContext private val context: Context,
        @NetworkInfo private val apiKey: String)// do the initialisation here
{

    val dummyData: String
        get() = "NETWORK_DUMMY_DATA"
}
